lib/sisimai/bite/email/gsuite.rb

```diff
@@ -0,0 +1,273 @@
+                v['rhost'] = cv[1].downcase
+                v['rhost'] = '' if v['rhost'] =~ /\A\s+\z/  # Remote-MTA: DNS;
+
+              elsif cv = e.match(/\A[Ll]ast-[Aa]ttempt-[Dd]ate:[ ]*(.+)\z/)
+                # Last-Attempt-Date: Fri, 24 Mar 2017 23:34:10 -0700 (PDT)
+                v['date'] = cv[1]
+
+              else
+                if cv = e.match(/\A[Dd]iagnostic-[Cc]ode:[ ]*(.+?);[ ]*(.+)\z/)
+                  # Diagnostic-Code: smtp; 550 #5.1.0 Address rejected.
+                  v['spec'] = cv[1].upcase
+                  v['diagnosis'] = cv[2]
+                else
+                  # Append error messages continued from the previous line
+                  if endoferror && ( v['diagnosis'] && v['diagnosis'].size > 0 )
+                    endoferror = true if e.empty?
+                    endoferror = true if e =~ /\A--/
+
+                    next if endoferror
+                    next unless e =~ /\A[ ]/
+                    v['diagnosis'] += e
+                  end
+                end
+              end
+
+            else
+              # Reporting-MTA: dns; googlemail.com
+              # Received-From-MTA: dns; sironeko@example.jp
+              # Arrival-Date: Fri, 24 Mar 2017 23:34:07 -0700 (PDT)
+              # X-Original-Message-ID: <06C1ED5C-7E02-4036-AEE1-AA448067FB2C@example.jp>
+              if cv = e.match(/\A[Rr]eporting-MTA:[ ]*(?:DNS|dns);[ ]*(.+)\z/)
+                # Reporting-MTA: dns; mx.example.jp
+                next if connheader['lhost'].size > 0
+                connheader['lhost'] = cv[1].downcase
+                connvalues += 1
+
+              elsif cv = e.match(/\A[Aa]rrival-[Dd]ate:[ ]*(.+)\z/)
+                # Arrival-Date: Wed, 29 Apr 2009 16:03:18 +0900
+                next if connheader['date'].size > 0
+                connheader['date'] = 1
+                connvalues += 1
+
+              else
+                # Detect SMTP session error or connection error
+                if e =~ Re1[:error]
+                  # The response from the remote server was:
+                  anotherset['diagnosis'] += e
+                else
+                  # ** Address not found **
+                  #
+                  # Your message wasn't delivered to * because the address couldn't be found.
+                  # Check for typos or unnecessary spaces and try again.
+                  #
+                  # The response from the remote server was:
+                  # 550 #5.1.0 Address rejected.
+                  next if e =~ Re1[:html]
+
+                  if anotherset['diagnosis']
+                    # Continued error messages from the previous line like
+                    # "550 #5.1.0 Address rejected."
+                    next if emptylines > 5
+                    if e.empty?
+                      # Count and next()
+                      emptylines += 1
+                      next
```

Comment:
__[rubocop]__
* Metrics/BlockNesting: Avoid more than 4 levels of block nesting. [:ref](https://github.com/bbatsov/ruby-style-guide#three-is-the-number-thou-shalt-count)

<!-- Comment by [SideCI](https://sideci.com) -->